### PR TITLE
test: Add auction integ tests

### PR DESCRIPTION
**Pull request by Eric-Warehime**
Closes BLO-585

Adding Network Suite tests for the auction lane.

Will continue adding iteratively to these--let me know feedback on edge cases you'd like to see added.

## Summary by CodeRabbit

- **Tests**
	- Introduced a new suite of integration tests for auction functionality, including bid validation and fee verification.
	- Updated network tests to account for changes in escrow account parameters during auctions.

### CODE REVIEW

File: tests/integration/network/auction_test.go [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+			auctiontypes.NewMsgAuctionBid(
+				bidder2.Address(),
+				params.Params.ReserveFee.Add(params.Params.MinBidIncrement),
+				[][]byte{send2Tx},
+			),
+		)
+		require.NoError(s.T(), err)
+
+		// Broadcast the bids
+		for _, tx := range [][]byte{bid1Tx, bid2Tx} {
```
Comment (nivasan1) [inline tests/integration/network/auction_test.go:170]: q why can't we synchronously broadcast both bids (BroadcastTxCommit)?

Comment (nivasan1) [reply tests/integration/network/auction_test.go:170]: nvm, we're expecting that bid2 is the only bid committed?

File: testutils/networksuite/networksuite.go [hunk 2]
```diff
@@ -81,6 +83,8 @@ func (nts *NetworkTestSuite) SetupSuite() {
 	// initialize genesis
 	require.NoError(nts.T(), cfg.Codec.UnmarshalJSON(cfg.GenesisState[auctiontypes.ModuleName], &nts.AuctionState))
 	nts.AuctionState = populateAuction(r, nts.AuctionState)
+	nts.AuctionEscrow = account.NewAccount()
+	nts.AuctionState.Params.EscrowAccountAddress = nts.AuctionEscrow.Address().Bytes()
```
Comment (nivasan1) [inline testutils/networksuite/networksuite.go:87]: why r we changing the escrow addy?

**Review by nivasan1 — APPROVED** [review]
lgtm

File: tests/integration/network/auction_test.go (hunk 1 above)
Comment (Eric-Warehime) [reply tests/integration/network/auction_test.go:170]: Yeah only one should actually commit.

File: testutils/networksuite/networksuite.go (hunk 2 above)
Comment (Eric-Warehime) [reply testutils/networksuite/networksuite.go:87]: For convenience of having an account object. I'll change it back because it does add some non-zero complication.
